scipy/special/specfun/specfun.f

```diff
@@ -10048,47 +10057,53 @@ SUBROUTINE JYZO(N,NT,RJ0,RJ1,RY0,RY1)
 15      X0=X
         CALL JYNDD(N,X,BJN,DJN,FJN,BYN,DYN,FYN)
         X=X-DJN/FJN
-        IF (X-X0.LT.-1) X=X0-1
-        IF (X-X0.GT.1) X=X0+1
         IF (DABS(X-X0).GT.1.0D-11) GO TO 15
-        IF (L.GE.1)THEN
-           IF (X.LE.RJ1(L)+0.5) THEN
-              X=XGUESS+PI
-              XGUESS=X
-              GO TO 15
-           ENDIF
-        END IF
         L=L+1
         RJ1(L)=X
-C       XXX: should have a better initial guess for large N ~> 100 here
-        X=X+PI+MAX((0.4955d0+0.0915*N-0.000435*N**2)/L, 0d0)
-        IF (L.LT.NT) GO TO 15
+        IF (L.LT.NT) THEN
+            X=X + (RJ0(L+1) - RJ0(L))
+            GO TO 15
```

Comment:
Similar to above, I think it would be good to cite https://dlmf.nist.gov/10.21#E20